DittoExportLogs/src/main/java/live/ditto/dittoexportlogs/ExportLogs.kt

```diff
@@ -1,26 +1,116 @@
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.res.stringResource
-import live.ditto.exporter.ExportDialog
-import java.nio.file.Path
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.launch
+import live.ditto.DittoError
+import live.ditto.DittoLogger
+import java.io.File
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
 
+@RequiresApi(Build.VERSION_CODES.O)
 @Composable
 fun ExportLogs(onDismiss: () -> Unit) {
-    ExportDialog(
-        title = stringResource(R.string.export_logs),
-        text = stringResource(R.string.do_you_want_to_export_logs),
-        confirmText = stringResource(R.string.export),
-        cancelText = stringResource(R.string.cancel),
-        fileProvider = { getZippedLogs().toFile() },
-        mimeType = stringResource(R.string.application_x_zip),
-        onDismiss = onDismiss
+
+    val title = stringResource(R.string.export_logs)
+
+    val confirmText = stringResource(R.string.export)
+    val cancelText = stringResource(R.string.cancel)
+    val modifier = Modifier
+    val coroutineScope = rememberCoroutineScope()
+    var isLoading by remember { mutableStateOf(false) }
+    val context = LocalContext.current
+
+    val now = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyyMMdd_HHmmss"))
+    val appName = context.applicationInfo.loadLabel(context.packageManager).toString()
+
+    val fileName = "$appName-dittologs-$now.jsonl.gz"
+    val downloadsDir = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS);
+    val filePath = File(downloadsDir, fileName).absolutePath
+
+    AlertDialog(
+        modifier = modifier,
+        onDismissRequest = onDismiss,
+        title = {
+            Text(title)
+        },
+        text = {
+            Text("Logs will be exported to: \nDownloads/$fileName\"")
+        },
+
+        confirmButton = {
+            Button(
+                onClick = {
+
+
+                    coroutineScope.launch {
+                        isLoading = true
+                        try {
+                            val sizeWritten = DittoLogger.exportToFile(filePath)
+
+                            val sizeWrittenInKB = sizeWritten / 1024u
+                            val toastText = "Exported file size: $sizeWrittenInKB kB"
+                            Toast.makeText(context, toastText, Toast.LENGTH_LONG).show()
+                        }
+                        catch (e: DittoError.IoError) {
+                            Log.e("DittoTools LogExporter", e.reason.toString())
+                            Toast.makeText(context, "Error exporting logs: ${e.message}", Toast.LENGTH_LONG).show()
+                        }
+                        finally {
+                            isLoading = false
+                            onDismiss()
+                        }
```

Comment:
Not sure what sort of time crunch we are on here, but I'd prefer this to be in a view model